tests/normalize/test_model_item_normalizer.py

```diff
@@ -0,0 +1,371 @@
+    monkeypatch.setenv("NORMALIZE__MODEL_NORMALIZER__ADD_DLT_ID", str(add_dlt_columns))
+    model_normalize = next(init_normalize())
+    # Define a query using the randomly ordered columns
+    cols = ", ".join(f"{col}" for col in columns)
+    model_query = f"SELECT {cols} FROM my_table"
+    dialect = caps.sqlglot_dialect
+
+    # Create a SQL model from the query
+    model = SqlModel.from_query_string(query=model_query, dialect=dialect)
+
+    # Ensure the schema contains the table "my_table" with columns from "a" to "d" appearing
+    # in alphabetical order
+    schema = create_schema_with_complete_columns("my_table", "text", ["a", "b", "c", "d"])
+
+    select_dialect, normalized_select_query, load_id = extract_normalize_retrieve(
+        model_normalize, model, schema, "my_table", dialect
+    )
+
+    assert select_dialect == dialect
+
+    # Parse the normalized model query to extract the aliases of the outer select
+    parsed_norm_select_query = sqlglot.parse_one(normalized_select_query, read=dialect)
+    assert isinstance(parsed_norm_select_query, sqlglot.exp.Select)
+    aliases = [select.alias for select in parsed_norm_select_query.selects]
+    if add_dlt_columns:
+        assert aliases == [
+            caps.casefold_identifier(col) for col in ["a", "b", "c", "d", "_dlt_load_id", "_dlt_id"]
+        ]
+    else:
+        assert aliases == [caps.casefold_identifier(col) for col in ["a", "b", "c", "d"]]
+
+
+@pytest.mark.parametrize("caps", MODEL_CAPS, indirect=True, ids=DESTINATIONS_SUPPORTING_MODEL)
+def test_select_column_added_from_schema(
+    caps: DestinationCapabilitiesContext, model_normalize: Normalize
+) -> None:
+    model_query = "SELECT a, b, c FROM my_table"
+    dialect = caps.sqlglot_dialect
+
+    # Create a SQL model from the query
+    model = SqlModel.from_query_string(query=model_query, dialect=dialect)
+
+    # Ensure the schema contains the table "my_table" with columns a, b, c and additionally d
+    schema = create_schema_with_complete_columns("my_table", "text", ["a", "b", "c", "d"])
+
+    select_dialect, normalized_select_query, _ = extract_normalize_retrieve(
+        model_normalize, model, schema, "my_table", dialect
+    )
+
+    assert select_dialect == dialect
+
+    # Parse the normalized model query to extract the aliases of the outer select
+    parsed_norm_select_query = sqlglot.parse_one(normalized_select_query, read=dialect)
+    assert isinstance(parsed_norm_select_query, sqlglot.exp.Select)
+    aliases = [select.alias for select in parsed_norm_select_query.selects]
+    assert aliases == [
+        caps.casefold_identifier(col) for col in ["a", "b", "c", "d", "_dlt_load_id", "_dlt_id"]
+    ]
+
+
+@pytest.mark.parametrize("caps", MODEL_CAPS, indirect=True, ids=DESTINATIONS_SUPPORTING_MODEL)
+def test_select_column_missing_in_schema(
+    caps: DestinationCapabilitiesContext, model_normalize: Normalize
+) -> None:
```

Comment:
This is a cool test / behavior, but I wonder if this will ever happen that we have a select where columns are missing in the schema. Anyway, let's keep it!